nltk/translate/util.py

```diff
@@ -0,0 +1,144 @@
+''' UTILITY Functions for ROUGE score for Machine Translation'''
+
+from __future__ import division
+
+def jacknifing(score_list, averaging=True):
+        ''' This is a averaging function for calculating ROUGE score
+    when multiple references are present. In this technique, from a 
+    given list of m values, m unique sets of m-1 elements is created.
+    Then the maximum is calculated from each of this m sets. The 
+    averaging is then performed on the m maximum values thus obtained.
+    This technique has been referred to as the Jacknifing technique 
+    in the original paper that introduces ROUGE-SCORE.
+
+
+    param score_list : list of scores over which averaging occurs
+    type (score_list) : list
+
+    param averaging : Jacknifing occurs if averaging is True
+    type (averaging) : boolean
+
+    >>> scores=[10, 20, 30, 40]
+    
+    >>> jacknifing(scores, averaging=False)
+    [10, 20, 30, 40]
+
+    >>> jacknifing(scores, averaging=True)
+    37.5
+
+    >>> scores =[10]
+    
+    >>> jacknifing(scores)
+    10.0
+
+
+    '''
+        if(len(score_list) == 1):
+            return sum(score_list)/len(score_list)
+        elif((len(score_list) > 1) and (averaging is False)):
+            return score_list
+        else:
+            '''
+            average : store the maximum scores
+            from the m different sets of m-1 scores.
+            such that m is the len(score_list)
+            '''
+            average = []
+            for i in score_list:
+                # dummy : list a particular combo of m-1 scores
+                dummy = [j for j in score_list if i != j]
+                average.append(max(dummy))
```

Comment:
`max()` should be able to take a generator, so do:

```
average.append(max(j for j in score_list if i != j])
```

Reply:
Wait a minute. `score_list` is in the outer and inner loop, wouldn't that always return `average = max(score_list)`?

Reply:
What is the purpose of this fallback?

Reply:
As the paper suggests , given a set of m scores from m different references , the jacknifing procedure creates m unique sets each having m-1 scores  and calculates the max from each of these m sets. All these m max. values thus obtained are finally averaged.

However this is not the same as **average = max(score_list)** as one of the elements in the list **average** is the second-max element from the **score_list**. This occurs when value of **i** in the outer loop is equal to the **max(score_list)**